### PR TITLE
Update linter rules for production release, version bump to 1.3.0

### PR DESCRIPTION
**Pull request by AkhilaIlla**
Followed steps mentioned: https://github.com/Azure/azure-openapi-validator/blob/main/CONTRIBUTING.md#create-a-release-pr

- Remove ```stagingOnly``` from all the signed off linter rules
- run ```rush version --bump```
- Deleted the unrelated changes made by the command, i.e, changes not related to the rulesets folder
- Added changelog

### CODE REVIEW

**Review by konrad-jamrozik — APPROVED** [review]
LGTM, but can you elaborate on who, how and when signed-off these rules?

File: packages/rulesets/generated/spectral/az-arm.js [hunk 1]
```diff
@@ -3145,8 +3143,7 @@ const ruleset = {
         },
         ValidateSegmentsInNestedResourceListOperation: {
```
Comment (rkmanda) [inline packages/rulesets/generated/spectral/az-arm.js:3144]: >ValidateSegmentsInNestedResourceListOperation [](http://example.com/codeflow?start=8&length=45)

Wasnt this missing the link to the md file? We should fix that before the upgrade

File: packages/rulesets/generated/spectral/az-arm.js [hunk 2]
```diff
@@ -3623,7 +3613,6 @@ const ruleset = {
             description: "This is a rule introduced to validate if provisioningState property is set to readOnly or not.",
             message: "{{error}}",
             severity: "warn",
```
Comment (rkmanda) [inline packages/rulesets/generated/spectral/az-arm.js:3615]: >warn [](http://example.com/codeflow?start=23&length=4)

error

**Review by rkmanda — CHANGES_REQUESTED** [review]
:clock1:

File: packages/rulesets/generated/spectral/az-arm.js (hunk 2 above)
Comment (AkhilaIlla) [reply packages/rulesets/generated/spectral/az-arm.js:3615]: Not sure why I has this as warn, weird.
Updated

File: packages/rulesets/generated/spectral/az-arm.js (hunk 1 above)
Comment (AkhilaIlla) [reply packages/rulesets/generated/spectral/az-arm.js:3144]: updated !
will move it to next release

**Review by rkmanda — CHANGES_REQUESTED** [review]
:clock1:

File: packages/rulesets/generated/spectral/az-arm.js (hunk 1 above)
Comment (rkmanda) [reply packages/rulesets/generated/spectral/az-arm.js:3144]: lets wait for Tejaswi's fix to get in and then remove this from staging before upgrading

**Pull request merged**